### PR TITLE
Tidy up CRIU support

### PR DESCRIPTION
**Pull request by keithc-ca**
* inline CRIU checkpoint state (only if needed)
* add missing `#endif` comments

### CODE REVIEW

File: runtime/vm/criuhelpers.cpp [hunk 1]
```diff
@@ -61,16 +61,17 @@ jvmCheckpointHooks(J9VMThread *currentThread)
 BOOLEAN
 jvmRestoreHooks(J9VMThread *currentThread)
 {
+#if defined(J9VM_OPT_CRIU_SUPPORT)
```
Comment (tajila) [inline runtime/vm/criuhelpers.cpp:64]: Can we put this entire function under a single `#ifdef`? There is no reason for this function here to be called if the CRIU flag is not enabled.

**Comment by tajila** [comment]
Adding Irwin to review the JIT changes

**Review by dsouzai — COMMENTED** [review]
No functional changes added to JIT, so LGTM for the JIT files.

File: runtime/vm/criuhelpers.cpp (hunk 1 above)
Comment (keithc-ca) [reply runtime/vm/criuhelpers.cpp:64]: Sure, I can do that (I was just maintaining the set of functions defined).

Comment (keithc-ca) [reply runtime/vm/criuhelpers.cpp]: The file is now only compiled if CRIU support is configured.

**Pull request merged**